main.tf

```diff
@@ -30,46 +30,56 @@ provider "kubernetes" {
 
 data "azurerm_subscription" "current" {}
 
-module "resource_group" {
-  source   = "./modules/azurerm_resource_group"
-  prefix   = var.prefix
-  name     = var.resource_group_name
+data "azurerm_resource_group" "network_rg" {
+  count    = var.vnet_resource_group_name == null ? 0 : 1
+  name     = var.vnet_resource_group_name
+}
+
+resource "azurerm_resource_group" "aks_rg" {
+  count    = var.resource_group_name == null ? 1 : 0
+  name     = "${var.prefix}-rg"
   location = var.location
   tags     = var.tags
 }
 
+data "azurerm_resource_group" "aks_rg" {
+  count    = var.resource_group_name == null ? 0 : 1
+  name     = var.resource_group_name
+}
 resource "azurerm_proximity_placement_group" "proximity" {
   count = var.node_pools_proximity_placement ? 1 : 0
 
   name                = "${var.prefix}-ProximityPlacementGroup"
   location            = var.location
-  resource_group_name = module.resource_group.name
-  tags                = module.resource_group.tags
-  depends_on          = [module.resource_group]
+  resource_group_name = local.aks_rg.name
+  tags                = var.tags
 }
 
-module "nsg" {
-  source              = "./modules/azurerm_network_security_group"
-  prefix              = var.prefix
-  name                = var.nsg_name
+resource "azurerm_network_security_group" "nsg" {
```

Comment:
Was the removal of the module for logic? Not sure why it was there and now it's been removed.

Reply:
removed the module for simplicity and consistency. With the additional vnet_rg it would have meant adding a couple more input/output vars to the module. That would have been more code than using the local/data/resource pattern.